app/poolChange.go

```diff
@@ -46,51 +45,27 @@ func GetPoolChangeFunc(app *SifchainApp) func(ctx sdk.Context, plan upgrade.Plan
 				temp = newLpUnits
 			}
 			pool.PoolUnits = temp
-			if !app.ClpKeeper.ValidatePool(pool) {
+			if !app.ClpKeeper.ValidatePool(*pool) {
 				hasError = true
 				ctx.Logger().Error(fmt.Sprintf("Invalid | Pool %s ", pool.String()))
 				break
 			}
-			poolList = append(poolList, pool)
+			poolList = append(poolList, *pool)
 		}
 		// If we have error dont set state
 		if hasError {
 			ctx.Logger().Error("Failed to execute upgrade plan for pool re-balance")
 		}
 		// If we have no errors , Set state .
 		if !hasError {
-			for _, pool := range poolList {
-				_ = app.ClpKeeper.SetPool(ctx, pool)
+			for i := range poolList {
+				pool := poolList[i]
+				_ = app.ClpKeeper.SetPool(ctx, &pool)
 			}
-			for _, l := range lps {
-				app.ClpKeeper.SetLiquidityProvider(ctx, l)
+			for i := range lps {
+				l := lps[i]
+				app.ClpKeeper.SetLiquidityProvider(ctx, &l)
 			}
 		}
 	}
-}
-
-func ExportAppState(name string, app *SifchainApp, ctx sdk.Context) {
```

Comment:
Any specific reason this was moved to a different file ? The reason this was kept here is becuase we wanted to keep and functionalities for a upgrade contained to one file. 
Just wondering if moving this provides some advantage ?